pkg/apis/serving/v1beta1/route_types.go

```diff
@@ -0,0 +1,139 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/knative/pkg/apis"
+	duckv1beta1 "github.com/knative/pkg/apis/duck/v1beta1"
+	"github.com/knative/pkg/kmeta"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Route is responsible for configuring ingress over a collection of Revisions.
+// Some of the Revisions a Route distributes traffic over may be specified by
+// referencing the Configuration responsible for creating them; in these cases
+// the Route is additionally responsible for monitoring the Configuration for
+// "latest ready" revision changes, and smoothly rolling out latest revisions.
+// See also: https://github.com/knative/serving/blob/master/docs/spec/overview.md#route
+type Route struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec holds the desired state of the Route (from the client).
+	// +optional
+	Spec RouteSpec `json:"spec,omitempty"`
+
+	// Status communicates the observed state of the Route (from the controller).
+	// +optional
+	Status RouteStatus `json:"status,omitempty"`
+}
+
+// Verify that Route adheres to the appropriate interfaces.
+var (
+	// Check that Route may be validated and defaulted.
+	_ apis.Validatable = (*Route)(nil)
+	_ apis.Defaultable = (*Route)(nil)
+
+	// Check that we can create OwnerReferences to a Route.
+	_ kmeta.OwnerRefable = (*Route)(nil)
+)
+
+// TrafficTarget holds a single entry of the routing table for a Route.
+type TrafficTarget struct {
+	// TODO(mattmoor): Add in whatever we end up calling "name"
+
+	// RevisionName of a specific revision to which to send this portion of traffic.
+	// This is mutually exclusive with ConfigurationName.
+	// +optional
+	RevisionName string `json:"revisionName,omitempty"`
+
+	// ConfigurationName of a configuration to whose latest revision we will send
+	// this portion of traffic. When the "status.latestReadyRevisionName" of the
+	// referenced configuration changes, we will automatically migrate traffic
+	// from the prior "latest ready" revision to the new one.
+	// This field is never set in Route's status, only its spec.
+	// This is mutually exclusive with RevisionName.
+	// +optional
+	ConfigurationName string `json:"configurationName,omitempty"`
+
+	// LatestRevision may be optionally provided to indicate that the latest
+	// ready Revision of the Configuration should be used for this traffic
+	// target.  When provided LatestRevision must be true if RevisionName is
+	// empty; it must be false when RevisionName is non-empty.
+	// +optional
+	LatestRevision *bool `json:"latestRevision,omitempty"`
+
+	// Percent specifies percent of the traffic to this Revision or Configuration.
+	// This defaults to zero if unspecified.
+	Percent int `json:"percent"`
```

Comment:
You have mentioned before that you thought this was `omitempty` in v1alpha1. Wanted to just raise it here to see if we should change here since all other fields are omitempty.

Also should have `+optional` above since this has a default behavior.

Reply:
Added